### PR TITLE
MAINT: fix typos

### LINKED ISSUE #27336: typos
from `ed` output

some content might be reworded
 
`combinaison`
 `many`
 `provides`
 `comptablity`
 `convenienes`
 `desirata`
 `maily`
 `obstable`
 `recution`
 `reductin`
<s>`reducting`</s>

```
$ ed -s numpy/numpy/_core/tests/test_numeric.py <<<'2181,2188p'
        """
        This test array_equal for a few combinaison:

        - are the two inputs the same object or not (same object many not
          be equal if contains NaNs)
        - Whether we should consider or not, NaNs, being equal.

        """
$ ed -s numpy/meson_cpu/main_config.h.in <<<'14p'
/// This definition is required to provides comptablity with NumPy distutils
$ ed -s numpy/doc/source/user/how-to-io.rst <<<'342,348p'
.. testcleanup::

   >>> import os
   >>> # list all files created in testsetup. If needed there are
   >>> # convenienes in e.g. astroquery to do this more automatically
   >>> for filename in ['csv.txt', 'fixedwidth.txt', 'nan.txt', 'skip.txt', 'tabs.txt']:
   ...     os.remove(filename)
$ ed -s numpy/numpy/_core/src/multiarray/buffer.c <<<'20,45p'
#include "dtypemeta.h"

/*************************************************************************
 ****************   Implement Buffer Protocol ****************************
 *************************************************************************/

/*************************************************************************
 * PEP 3118 buffer protocol
 *
 * Implementing PEP 3118 is somewhat convoluted because of the desirata:
 *
 * - Don't add new members to ndarray or descr structs, to preserve binary
 *   compatibility. (Also, adding the items is actually not very useful,
 *   since mutability issues prevent an 1 to 1 relationship between arrays
 *   and buffer views.)
 *
 * - Don't use bf_releasebuffer, because it prevents PyArg_ParseTuple("s#", ...
 *   from working. Breaking this would cause several backward compatibility
 *   issues already on Python 2.6.
 *
 * - Behave correctly when array is reshaped in-place, or it's dtype is
 *   altered.
 *
 * The solution taken below is to manually track memory allocated for
 * Py_buffers.
 *************************************************************************/
$ ed -s numpy/numpy/_core/tests/test_ufunc.py <<<'2547,2549p'
        # Checks the case where the we have a simple byte-swap works, maily
        # tests that this is not rejected directly.
        # (interesting because we require descriptor identity in reducelikes).
$ ed -s numpy/doc/source/building/cross_compilation.rst <<<'25,29p'
One possible hiccup is that the build requires running a compiled executable in
order to determine the ``long double`` format for the host platform. This may be
an obstable, since it requires ``crossenv`` or QEMU to run the host (cross)
Python. To avoid this problem, specify the paths to the relevant directories in
your *cross file*:
$ ed -s numpy/numpy/lib/recfunctions.py <<<'1212,1215p'
    Returns
    -------
    out : ndarray
       Result of the recution operation
$ ed -s numpy/numpy/_core/tests/test_custom_dtypes.py <<<'119,120p'
        # Addition reductin works (as of writing requires to pass initial
        # because setting a scaled-float from the default `0` fails).
$ 
```

@rkern

> file is vendored

```
$ ed -s numpy/numpy/random/src/pcg64/pcg64.orig.h <<<'767,769p'
 *     Empricical tests show that division is preferable to modulus for
 *     reducting the range of an RNG.  It's faster, and sometimes it can
 *     even be statistically preferable.
$ 
```

don't know if `reprecated` factually was intended here

```
$ ed -s numpy/numpy/lib/__init__.py <<<'56p'
    # Warn for reprecated attributes
$
```

a few numpy/linalg/lapack_lite directory typos

might be eliminated by regenerating those files with a newer lapack version

however note the following, lapack 3.2.2 was released on June 30, 2010

whereas lapack 3.12.0 (current stable) was released on November 24, 2023

```
$ tail -n 6 numpy/numpy/linalg/lapack_lite/README.rst
The output C files in git use the LAPACK source from the LAPACK_ page, using
version 3.2.2. Unfortunately, newer versions use newer FORTRAN features, which
are increasingly not supported by ``f2c``. As these are found, the patch files
will need to be changed to re-express new constructs with legacy constructs.

.. _LAPACK: https://netlib.org/lapack/index.html
$ 
```

<hr>

the following shell script is authorised for use and/or modification by the repository Maintainers

@seiko2plus 

a commented out line includes the string `whiher`

```
$ cat typos.sh
#!/bin/sh

sed -i "s/Incremement/Increment/g" numpy/doc/source/reference/c-api/array.rst
sed -i "s/Universell/Universal/g" numpy/doc/neps/nep-0021-advanced-indexing.rst
sed -i "s/absuolate/absolute/g" numpy/doc/source/reference/simd/how-it-works.rst
sed -i "s/accomblishable/accomplishable/g" numpy/doc/source/release/1.10.0-notes.rst
sed -i "s/ambiguuous/ambiguous/g" numpy/numpy/_core/tests/test_arrayprint.py
sed -i "s/andtheir/and their/g" numpy/doc/source/f2py/f2py-testing.rst
sed -i "s/assyemetric/asymmetric/g" numpy/numpy/_core/src/umath/scalarmath.c.src
sed -i "s/brach/branch/g" numpy/.spin/cmds.py
sed -i "s/combain/combine/g" numpy/doc/source/reference/simd/build-options.rst
sed -i "s/combintations/combinations/g" numpy/numpy/_core/tests/test_dtype.py
sed -i "s/comefrom/come from/g" numpy/numpy/_core/src/umath/ufunc_object.c
sed -i "s/complelx/complex/g" numpy/numpy/_core/tests/test_scalarmath.py
sed -i "s/concatentor/concatenator/g" numpy/numpy/lib/_index_tricks_impl.py
sed -i "s/conraction/contraction/g" numpy/numpy/_core/tests/test_einsum.py
sed -i "s/constructings/constructing/g" numpy/doc/source/reference/c-api/datetimes.rst
sed -i "s/contiuguous/contiguous/g" numpy/numpy/_core/tests/test_indexing.py
sed -i "s/customizatoin/customization/g" numpy/numpy/_core/src/multiarray/dtypemeta.h
sed -i "s/discourged/discouraged/g" numpy/numpy/_typing/_dtype_like.py
sed -i "s/ eaders/ headers/g" numpy/doc/neps/nep-0053-c-abi-evolution.rst
sed -i "s/extensino/extension/g" numpy/numpy/distutils/tests/test_build_ext.py
sed -i "s/functionalilty/functionality/g" numpy/doc/source/user/c-info.python-as-glue.rst
sed -i "s/hetergeneous/heterogeneous/g" numpy/numpy/lib/tests/test_loadtxt.py
sed -i "s/increadibly/incredibly/g" numpy/numpy/_core/src/multiarray/abstractdtypes.c
sed -i "s/inheritcs/inherits/g" numpy/numpy/__init__.pyi
sed -i "s/interbally/internally/g" numpy/numpy/lib/recfunctions.py
sed -i "s/intrinics/intrinsics/g" numpy/numpy/_core/src/umath/loops_arithm_fp.dispatch.c.src
sed -i "s/leasat/least/g" numpy/numpy/_core/src/common/numpyos.c
sed -i "s/nessecery/necessary/g" numpy/numpy/_core/src/common/npy_cpu_features.c
sed -i "s/obsolute/obsolete/g" numpy/numpy/f2py/diagnose.py
sed -i "s/obsolute/obsolete/g" numpy/numpy/f2py/capi_maps.py
sed -i "s/organizating/organizing/g" numpy/doc/neps/nep-0048-spending-project-funds.rst
sed -i "s/ortherwise/organizing/g" numpy/doc/neps/nep-0048-spending-project-funds.rst
sed -i "s/ortherwise/otherwise/g" numpy/numpy/distutils/mingw32ccompiler.py
sed -i "s/paramteric/parametric/g" numpy/numpy/_core/tests/test_casting_unittests.py
sed -i "s/partititioned/partitioned/g" numpy/numpy/_core/src/common/dlpack/dlpack.h
sed -i "s/quarantees/guarantees/g" numpy/numpy/_core/src/umath/fast_loop_macros.h
sed -i "s/ quivalent/ equivalent/g" numpy/numpy/typing/mypy_plugin.py
sed -i "s/relevants/relevant/g" numpy/numpy/distutils/exec_command.py
sed -i "s/respecitely/respectively/g" numpy/numpy/f2py/src/fortranobject.c
sed -i "s/restrored/restored/g" numpy/numpy/_core/tests/test_arrayprint.py
sed -i "s/rewinded/rewound/g" numpy/benchmarks/benchmarks/bench_io.py
sed -i "s/routies/routines/g" numpy/doc/source/user/basics.creation.rst
sed -i "s/runninng/running/g" numpy/numpy/_core/_machar.py
sed -i "s/samilar/similar/g" numpy/numpy/_core/_add_newdocs.py
sed -i "s/sometimed/sometimes/g" numpy/numpy/_core/fromnumeric.py
sed -i "s/suffux/suffix/g" numpy/doc/source/reference/simd/how-it-works.rst
sed -i "s/supppressed/suppressed/g" numpy/numpy/testing/tests/test_utils.py
sed -i "s/underlyign/underlying/g" numpy/numpy/random/mtrand.pyx
sed -i "s/unifing/unifying/g" numpy/numpy/distutils/command/config_compiler.py
sed -i "s/unvavailable/unavailable/g" numpy/numpy/random/src/mt19937/randomkit.c
# sed -i "s/whiher/whether/g" numpy/meson_cpu/main_config.h.in
sed -i "s/widthes/widths/g" numpy/numpy/_core/tests/test_multiarray.py
sed -i "s/worstcase/worst-case/g" numpy/numpy/_core/src/npysort/selection.cpp
$ 
```

_modifications_

removed from shell script

no `workflow` scope unable to modify the .yml file

```
sed -i "s/intrincis/intrinsics/g" numpy/.github/workflows/linux_simd.yml
```

@rkern

and three other lines have been removed from the shell script

> file is vendored

```
sed -i "s/Empricical/Empirical/g" numpy/numpy/random/src/pcg64/pcg64.orig.h
```

source

```
$ git clone -q https://github.com/imneme/pcg-c.git
$ grep -nr Empricical pcg-c
pcg-c/include/pcg_variants.h:865: *     Empricical tests show that division is preferable to modulus for
$ 
```

> style of quote in the rest of the file

```
sed -i "s/shouldn’t/shouldn't/g" numpy/doc/neps/nep-0022-ndarray-duck-typing-overview.rst
```

> form is acceptable

```
sed -i "s/unpickleable/unpicklable/g" numpy/numpy/_core/tests/test_regression.py
```

### PR DESCRIPTION
**Pull request by musvaage**
Closes https://github.com/numpy/numpy/issues/27336

### CODE REVIEW

File: doc/neps/nep-0022-ndarray-duck-typing-overview.rst
Comment (rkern) [inline doc/neps/nep-0022-ndarray-duck-typing-overview.rst:141]: This should be left alone.

File: numpy/_core/tests/test_regression.py
Comment (rkern) [inline numpy/_core/tests/test_regression.py:456]: Leave this one alone. Either form is acceptable.

File: numpy/random/src/pcg64/pcg64.orig.h
Comment (rkern) [inline numpy/random/src/pcg64/pcg64.orig.h:767]: This file is vendored and should be left alone to match the upstream version.

**Comment by rkern** [comment]
I've commented on the ones that should be left alone. The others seem to be correct fixes, spelling-wise, though I haven't verified if any have practical considerations (e.g. if changing a warning message might have consequences for things that try to catch that warning message, etc.), but I doubt there are many of those.

File: doc/neps/nep-0022-ndarray-duck-typing-overview.rst
Comment (musvaage) [reply doc/neps/nep-0022-ndarray-duck-typing-overview.rst:141]: not per se a typo fix replacing U+2019 with U+0027 but pursued as it is easier to deal with dumped strings

if you still think it should be left alone it can easily be reverted

File: numpy/_core/tests/test_regression.py
Comment (musvaage) [reply numpy/_core/tests/test_regression.py:456]: it was changed in pandas

but again if you still think it should be left alone it can easily be reverted

File: doc/neps/nep-0022-ndarray-duck-typing-overview.rst
Comment (rkern) [reply doc/neps/nep-0022-ndarray-duck-typing-overview.rst:141]: Yes, please leave it alone. We use that style of quote in the rest of the file.

File: numpy/_core/tests/test_regression.py
Comment (rkern) [reply numpy/_core/tests/test_regression.py:456]: https://en.wiktionary.org/wiki/pickleable

Yes, please leave it alone.

File: numpy/random/src/pcg64/pcg64.orig.h
Comment (musvaage) [reply numpy/random/src/pcg64/pcg64.orig.h]: pcg-c/include/pcg_variants.h

**Comment by rkern** [comment]
Please don't force-push for PRs like this. It makes it very difficult to keep track of which changes have been reviewed and which still need review.

**Comment by musvaage** [comment]
numpy/_core/tests/test_casting_unittests.py

presuming modifying this file is inconsequential

personally my inclination is to merge this commit which implements solely those fixes from the ticket's shell script

the ticket's `ed` output could later be revisited and a second pull made

likewise with a third pull were a Maintainer decision to be made

on whether to regenerate numpy/linalg/lapack_lite directory files with a newer lapack version

**Comment by musvaage** [comment]
> force-push

your requested changes were implemented and a soft reset was performed

**Comment by musvaage** [comment]
> from ef7b488

of course the force-pushed affected files are seen in the above

**Comment by rkern** [comment]
The lint checks need to be fixed before merging.

**Comment by mattip** [comment]
Since the linting errors are in the deprecated distutils module, I don't think they should be fixed.

**Review by mattip — APPROVED** [review]
I looked through these and they all seem reasonable, including the warning/error message fixes.

**Pull request merged**
